src/vm/hints/find_element_hint.rs

```diff
@@ -0,0 +1,558 @@
+    let (array_ptr_ref, elm_size_ref, n_elms_ref, index_ref, key_ref) = if let (
+        Some(array_ptr_ref),
+        Some(elm_size_ref),
+        Some(n_elms_ref),
+        Some(index_ref),
+        Some(key_ref),
+    ) = (
+        ids.get("array_ptr"),
+        ids.get("elm_size"),
+        ids.get("n_elms"),
+        ids.get("index"),
+        ids.get("key"),
+    ) {
+        (array_ptr_ref, elm_size_ref, n_elms_ref, index_ref, key_ref)
+    } else {
+        return Err(VirtualMachineError::IncorrectIds(
+            vec!["array_ptr", "elm_size", "n_elms", "index", "key"]
+                .iter()
+                .map(|s| s.to_string())
+                .collect(),
+            ids.into_keys().collect(),
+        ));
+    };
+
+    let (array_ptr_addr, elm_size_addr, n_elms_addr, index_addr, key_addr) = if let (
+        Ok(Some(array_ptr_addr)),
+        Ok(Some(elm_size_addr)),
+        Ok(Some(n_elms_addr)),
+        Ok(Some(index_addr)),
+        Ok(Some(key_addr)),
+    ) = (
+        get_address_from_reference(
+            array_ptr_ref,
+            &vm.references,
+            &vm.run_context,
+            vm,
+            hint_ap_tracking,
+        ),
+        get_address_from_reference(
+            elm_size_ref,
+            &vm.references,
+            &vm.run_context,
+            vm,
+            hint_ap_tracking,
+        ),
+        get_address_from_reference(
+            n_elms_ref,
+            &vm.references,
+            &vm.run_context,
+            vm,
+            hint_ap_tracking,
+        ),
+        get_address_from_reference(
+            index_ref,
+            &vm.references,
+            &vm.run_context,
+            vm,
+            hint_ap_tracking,
+        ),
+        get_address_from_reference(
+            key_ref,
+            &vm.references,
+            &vm.run_context,
+            vm,
+            hint_ap_tracking,
+        ),
+    ) {
+        (
+            array_ptr_addr,
+            elm_size_addr,
+            n_elms_addr,
+            index_addr,
+            key_addr,
+        )
+    } else {
+        return Err(VirtualMachineError::FailedToGetIds);
+    };
+
+    match (
+        vm.memory.get(&array_ptr_addr),
+        vm.memory.get(&elm_size_addr),
+        vm.memory.get(&n_elms_addr),
+        vm.memory.get(&index_addr),
+        vm.memory.get(&key_addr),
+    ) {
+        (
+            Ok(_),
+            Ok(Some(maybe_rel_elm_size)),
+            Ok(Some(maybe_rel_n_elms)),
+            Ok(_),
+            Ok(Some(maybe_rel_key)),
+        ) => {
+            for (name, builtin) in &vm.builtin_runners {
+                //Check that range_check_builtin is present
+                let _builtin = match builtin.as_any().downcast_ref::<RangeCheckBuiltinRunner>() {
+                    Some(b) => b,
+                    None => return Err(VirtualMachineError::NoRangeCheckBuiltin),
+                };
+
+                if name == &"range_check".to_string() {
```

Comment:
This part can become (will need to update everything below):
```suggestion
            let builtin = vm.builtin_runners.iter()
                .find(|(name, _)| name.as_str() == "range_check")
                .ok_or(VirtualMachineError::NoRangeCheckBuiltin)?
                .1
                .as_any().downcast_ref::<RangeCheckBuiltinRunner>()
                .map_err(VirtualMachineError::NoRangeCheckBuiltin)?;
```

Reply:
resolved: https://github.com/lambdaclass/cleopatra_cairo/pull/294/commits/25fad2bc0eb02e2dd81e354dea1e506c6c2373b5